gui/src/JobSelector.tsx

```diff
@@ -0,0 +1,711 @@
+                            </IconButton>
+                        </Grid>
+                    </Grid>
+                    <Grid item xs={12} container justifyContent="space-between" mt={2}>
+                        <Button variant="outlined" color="inherit" onClick={removeAllFilter}>
+                            REMOVE ALL FILTERS
+                        </Button>
+                        <Button variant="contained" type="submit">
+                            APPLY FILTER
+                        </Button>
+                    </Grid>
+                </Grid>
+            </form>
+        </Popover>
+    );
+};
+
+interface JobSelectedTableHeadProps {
+    selectAllTrue: boolean;
+    handleSelectAllClick?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+    columns: JobSelectorColumns[];
+}
+function JobSelectedTableHead({ selectAllTrue, handleSelectAllClick, columns }: JobSelectedTableHeadProps) {
+    return (
+        <TableHead>
+            <TableRow>
+                <TableCell padding="checkbox">
+                    <Checkbox color="primary" checked={selectAllTrue} onChange={handleSelectAllClick} />
+                </TableCell>
+                {columns
+                    .filter((i) => i.showPrimaryLabel || i.showSecondayLabel)
+                    .map((item, index) => {
+                        return (
+                            <TableCell key={index}>
+                                {item.secondaryLabel ? (
+                                    <ListItemText primary={item.primaryLabel} secondary={item.secondaryLabel} />
+                                ) : (
+                                    item.primaryLabel
+                                )}
+                            </TableCell>
+                        );
+                    })}
+                <TableCell>Actions</TableCell>
+            </TableRow>
+        </TableHead>
+    );
+}
+
+interface JobSelectedTableRowProps {
+    row: Job;
+    isItemSelected: boolean;
+    handleCheckboxClick: (event: React.MouseEvent<HTMLElement>, name: string) => void;
+    handleCancelJobs: (id: string[]) => void;
+    handleDeleteJobs: (id: string[]) => void;
+    displayJobId?: boolean;
+    displayEntityLabel?: boolean;
+    displayEntityId?: boolean;
+    displaySubmitId?: boolean;
+    displayDate?: boolean;
+}
+function JobSelectedTableRow({
+    row,
+    isItemSelected,
+    handleCheckboxClick,
+    handleCancelJobs,
+    handleDeleteJobs,
+    displayJobId,
+    displayEntityLabel,
+    displayEntityId,
+    displaySubmitId,
+    displayDate,
+}: JobSelectedTableRowProps) {
+    const [id, jobName, entityId, entityName, submitId, creationDate, status] = row;
+
+    const doDeleteJob = useCallback(
+        (id: string[]) => {
+            handleDeleteJobs && handleDeleteJobs(id);
+        },
+        [handleDeleteJobs]
+    );
+    const doCancelJob = useCallback(
+        (id: string[]) => {
+            handleCancelJobs && handleCancelJobs(id);
```

Comment:
I don't see the need of redefining a local callback. why not use the property directly (fetching the id by data-id) ?